src/components/DatButton.vue

```diff
@@ -1,19 +1,31 @@
 <script setup lang="ts">
+import { computed } from "vue";
+
 interface Props {
   label?: string;
+  disabled?: boolean;
 }
 
-withDefaults(defineProps<Props>(), {
+const props = withDefaults(defineProps<Props>(), {
   label: "",
+  disabled: false,
+});
+const watchDisabled = computed(() => {
+  return props.disabled;
 });
 const emit = defineEmits<{
   click: [value: MouseEvent];
 }>();
-const handleClick = (event: MouseEvent) => emit("click", event);
+const handleClick = (event: MouseEvent) => {
+  if (watchDisabled.value) {
+    return;
+    emit("click", event);
+  }
```

Comment:
The return needs to happen only if `watchDisabled.value` is true, otherwise we won't emit anymore.
And same comment as before, rename `watchDisabled` to `isDisabled`

```suggestion
  if (watchDisabled.value) return;
  
  emit("click", event);
```